### ISSUE TITLE
Allowed Extensions

### ISSUE THREAD

**Issue by webclubz** [getformwork/formwork#57]
First i want to thank you for this great flexible cms.  Just something to mention...
When adding new file extensions in Options/Allowed Extensions eg .zip and trying to upload a file with this extension getting an admin error in all pages. It works fine only if also add extra extensions manually in formwork/Files/File.php ` if (in_array($this->extension, ['pdf', 'doc', 'docx', 'odt', 'zip', '7z',.......

Thank you again, you have done excellent work

**Comment by giuscris** [comment]
Hello @webclubz thank you for using Formwork.
Can you give me more context about the issue? What error do you get exactly?

**Comment by giuscris** [comment]
I have understood what is the issue: `File::type()` has the wrong return type.
But I couldn’t replicate the error because it's been fixed for the next release (581a66ee969590b1157953b54ad83a3f79b7b42b).

In the meantime, before the release of 10.11.0, you can fix this by yourself, just change the function declaration in `formwork/Files/File.php` from this:

https://github.com/getformwork/formwork/blob/62e336ddfa989e044a9f8e9d5ca30e63240398d9/formwork/Files/File.php#L119

To this:

https://github.com/getformwork/formwork/blob/d05a0de1cb9e6fed93fe4eeead9e24de429480ff/formwork/Files/File.php#L119

Basically you have to change `: string` to `: ?string` at the end of the line.
I’m sure this fixes the problem for you before the next release, which will fix this anyway.

**Comment by webclubz** [comment]
Yep that works... Thank you Giuseppe!
